### PR TITLE
[fix] Check the library in advance to avoid program exit without using NativeIO

### PR DESCRIPTION
**Pull request by wenbingshen**
### Motivation

Related to this PR: https://github.com/apache/bookkeeper/pull/3824

We should check whether the corresponding library is exist in advance, otherwise an error will be reported without using NativeIO.

Below is the error I am getting:
```
2023-03-02T16:48:13,476 - ERROR - [ForceWriteThread:BookieCriticalThread@41] - Uncaught exception in thread ForceWriteThread and is exiting!
java.lang.NoClassDefFoundError: Could not initialize class org.apache.bookkeeper.common.util.nativeio.NativeIOJni
	at org.apache.bookkeeper.common.util.nativeio.NativeIOImpl.posix_fadvise(NativeIOImpl.java:45) ~[classes/:?]
	at org.apache.bookkeeper.util.PageCacheUtil.bestEffortRemoveFromPageCache(PageCacheUtil.java:97) ~[classes/:?]
	at org.apache.bookkeeper.bookie.JournalChannel.forceWrite(JournalChannel.java:326) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteRequest.process(Journal.java:402) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteThread.processForceWriteRequest(Journal.java:597) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteThread.run(Journal.java:539) ~[classes/:?]
2023-03-02T16:48:13,476 - ERROR - [ForceWriteThread:BookieCriticalThread@41] - Uncaught exception in thread ForceWriteThread and is exiting!
java.lang.NoClassDefFoundError: Could not initialize class org.apache.bookkeeper.common.util.nativeio.NativeIOJni
	at org.apache.bookkeeper.common.util.nativeio.NativeIOImpl.posix_fadvise(NativeIOImpl.java:45) ~[classes/:?]
	at org.apache.bookkeeper.util.PageCacheUtil.bestEffortRemoveFromPageCache(PageCacheUtil.java:97) ~[classes/:?]
	at org.apache.bookkeeper.bookie.JournalChannel.forceWrite(JournalChannel.java:326) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteRequest.process(Journal.java:402) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteThread.processForceWriteRequest(Journal.java:597) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteThread.run(Journal.java:539) ~[classes/:?]
2023-03-02T16:48:13,476 - ERROR - [ForceWriteThread:BookieCriticalThread@41] - Uncaught exception in thread ForceWriteThread and is exiting!
java.lang.ExceptionInInitializerError: null
	at org.apache.bookkeeper.common.util.nativeio.NativeIOImpl.posix_fadvise(NativeIOImpl.java:45) ~[classes/:?]
	at org.apache.bookkeeper.util.PageCacheUtil.bestEffortRemoveFromPageCache(PageCacheUtil.java:97) ~[classes/:?]
	at org.apache.bookkeeper.bookie.JournalChannel.forceWrite(JournalChannel.java:326) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteRequest.process(Journal.java:402) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteThread.processForceWriteRequest(Journal.java:597) ~[classes/:?]
	at org.apache.bookkeeper.bookie.Journal$ForceWriteThread.run(Journal.java:539) ~[classes/:?]
Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: Couldn't find file into jar /lib/libnative-io.jnilib
	at org.apache.bookkeeper.common.util.nativeio.NativeIOJni.<clinit>(NativeIOJni.java:61) ~[classes/:?]
	... 6 more
Caused by: java.io.FileNotFoundException: Couldn't find file into jar /lib/libnative-io.jnilib
	at org.apache.bookkeeper.common.util.nativeio.NativeUtils.loadLibraryFromJar(NativeUtils.java:66) ~[classes/:?]
	at org.apache.bookkeeper.common.util.nativeio.NativeIOJni.<clinit>(NativeIOJni.java:54) ~[classes/:?]
	... 6 more

```

### CODE REVIEW

**Comment by wenbingshen** [comment]
rerun failure checks

**Review by horizonzy — COMMENTED** [review]
It's indeed a problem. The test can't pass due to the idea can't find the `/lib/libnative-io.jnilib` file. The idea only finds the resource under `/native-io/target/classes`, the `/lib/libnative-io.jnilib` didn't package into `/native-io/target/classes`.

But if you use mvn command to run the test, it passes.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/util/PageCacheUtil.java [hunk 1]
```diff
@@ -34,22 +36,39 @@ public final class PageCacheUtil {
...
 
-    private static boolean fadvisePossible = true;
+    private static boolean fadvisePossible = false;
 
     private static final NativeIO NATIVE_IO;
 
     static {
         NativeIO nativeIO = null;
         try {
+            if (SystemUtils.IS_OS_MAC_OSX) {
```
Comment (horizonzy) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/util/PageCacheUtil.java:46]: Moving the check logic in NativeIOImpl construction will be better. The NativeIOImpl is strongly bound with lib.

**Comment by horizonzy** [comment]
It can be closed. Catching the throwable will cover the exception(`java.lang.ExceptionInInitializerError`).

**Comment by wenbingshen** [comment]
> It can be closed. Catching the throwable will cover the exception(`java.lang.ExceptionInInitializerError`).

@horizonzy Good catch. Your method looks better. I didn't notice this detail yesterday. You can submit a PR to fix it and I will close this PR. :)

**Comment by horizonzy** [comment]
Matteo will fix it.

**Pull request closed without merging**